roles/grafana/tasks/preflight.yml

```diff
@@ -42,12 +42,14 @@
           ( grafana_database.type == 'sqlite3' and grafana_database.password is defined ) or
           ( grafana_database.type == 'sqlite3' and grafana_database.server_cert_name is defined )"
 
-- name: "Fail when grafana domain isn't properly configured"
+- name: "Fail when grafana_server isn't properly configured"
   ansible.builtin.fail:
-    msg: "Check server configuration. Please look at http://docs.grafana.org/installation/configuration/#server"
+    msg: "Wrong grafana_server configuration parameter. Please look at https://github.com/grafana/grafana-ansible-collection/blob/main/README.md"
   when:
-    - "grafana_server.root_url is defined"
-    - "grafana_server.root_url is search(grafana_server.domain)"
+    "grafana_server.http_addr is defined or
+    grafana_server.http_port is defined or
+    grafana_server.domain is defined or
+    grafana_server.root_url is defined"
```

Comment:
Looks good to me !

Reply:
Wrote that from memory, were you able to verify that it actually works?

Reply:
hi, I assumed initially you did that :). The reason I didn't change to assertion because of not having used it before and it looked good with the previous change. With some hassle I was able to test it, and it works.
Output looks like :
TASK [ansible_icinga2_grafana_director_setup : Fail when grafana_server isn't properly configured] **************************************************************************
failed: [icinga2test.binero.net] (item={'option': 'http_addr', 'role_var': 'grafana_address'}) => {
    "ansible_loop_var": "item",
    "assertion": "item.option not in grafana_server",
    "changed": false,
    "evaluated_to": false,
    "item": {
        "option": "http_addr",
        "role_var": "grafana_address"
    },
    "msg": "Invalid grafana_server configuration parameter: http_addr. Use variable grafana_address instead"
}
failed: [icinga2test.binero.net] (item={'option': 'http_port', 'role_var': 'grafana_port'}) => {
    "ansible_loop_var": "item",
    "assertion": "item.option not in grafana_server",
    "changed": false,
    "evaluated_to": false,
    "item": {
        "option": "http_port",
        "role_var": "grafana_port"
    },
    "msg": "Invalid grafana_server configuration parameter: http_port. Use variable grafana_port instead"
}
failed: [icinga2test.binero.net] (item={'option': 'domain', 'role_var': 'grafana_domain'}) => {
    "ansible_loop_var": "item",
    "assertion": "item.option not in grafana_server",
    "changed": false,
    "evaluated_to": false,
    "item": {
        "option": "domain",
        "role_var": "grafana_domain"
    },
    "msg": "Invalid grafana_server configuration parameter: domain. Use variable grafana_domain instead"
}
ok: [icinga2test.binero.net] => (item={'option': 'root_url', 'role_var': 'grafana_url'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": {
        "option": "root_url",
        "role_var": "grafana_url"
    },
    "msg": "All assertions passed"
}

after I added the item.option too, I updated the PR.
Looks pretty neat indeed, although I was slightly happier with the previous, also because it pointed to the README. But this is fine with me :).